In [1]:
import pandas as pd

In [2]:
airbnb = pd.read_csv('airbnb_data.csv')

In [3]:
%%sql
SELECT AVG(num_listings) AS avg_listings_per_host
FROM (
    SELECT host_id, COUNT(DISTINCT listing_id) AS num_listings
    FROM airbnb
    GROUP BY host_id
) AS host_listings;

avg_listings_per_host
0               1.101082

In [4]:
%%sql
SELECT 
    CORR(avg_rating, num_listings) AS correlation_rating_listings,
    CORR(avg_price, num_listings) AS correlation_price_listings
FROM (
    SELECT 
        host_id,
        AVG(rating) AS avg_rating,
        AVG(price) AS avg_price,
        COUNT(DISTINCT listing_id) AS num_listings
    FROM 
        airbnb
    GROUP BY 
        host_id
) AS host_data;

correlation_rating_listings  correlation_price_listings
0                     0.002957                    0.003868

Average Number of Listings per Host:

- Calculated the average number of listings per host in the dataset.
- Counted the distinct number of listings associated with each host and then calculated the average across all hosts.
- Result: The average number of listings per host is approximately 1.10.

Correlation between Number of Listings and Average Rating or Price:

- Investigated whether there is a correlation between the number of listings and the average rating or price per host.
- Calculated the correlation coefficients between the number of listings and the average rating or price.
- Results: 
  - Correlation between the number of listings and average rating: 0.002957
  - Correlation between the number of listings and average price: 0.003868
- These values indicate that there is almost no linear relationship between the number of listings and either the average rating or the average price per host in the dataset.